orthoinference/src/main/java/org/reactome/orthoinference/StableIdentifierGenerator.java

```diff
@@ -1,67 +1,73 @@
 package org.reactome.orthoinference;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.gk.model.GKInstance;
 import org.gk.persistence.MySQLAdaptor;
 
+import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+
 import static org.gk.model.ReactomeJavaConstants.*;
 
 /*
  *  All PhysicalEntitys, ReactionlikeEvents and Pathways are routed to this class to generate their stable identifiers
  */
 public class StableIdentifierGenerator {
-
-    private static MySQLAdaptor dba;
-    private static GKInstance instanceEditInst;
+    private static final Logger logger = LogManager.getLogger();
     private static Map<String,Integer> seenOrthoIds = new HashMap<>();
-    private static String speciesAbbreviation = null;
 
-    public static GKInstance generateOrthologousStableId(GKInstance inferredInst, GKInstance originalInst) throws Exception {
+    private MySQLAdaptor dba;
+    private String speciesAbbreviation;
+
+    public StableIdentifierGenerator(MySQLAdaptor dba, String speciesAbbreviation) {
+        this.dba = dba;
+        this.speciesAbbreviation = speciesAbbreviation;
+    }
+
+    public GKInstance generateOrthologousStableId(GKInstance inferredInst, GKInstance originalInst) throws Exception {
 
         // All Human PhysicalEntitys and Events will have a StableIdentifier instance in the stableIdentifier attribute
         GKInstance stableIdentifierInst = (GKInstance) originalInst.getAttributeValue(stableIdentifier);
+        if (stableIdentifierInst == null) {
+            logger.fatal("No stable identifier instance found for " + originalInst + " -- terminating");
+            throw new RuntimeException("Could not find stable identifier instance");
```

Comment:
It might be good to keep the logging message and the exception message the same.  The log statement is nice since it gives the instance that didn't have a stable id, but maybe the "terminating" part can be removed in case the exception is ever caught and dealt with so that the program keeps going?

Reply:
I removed the terminating part and made it the same message for the logger and exception message